model/visor/processing.go

```diff
@@ -0,0 +1,201 @@
+package visor
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/filecoin-project/lotus/chain/types"
+	"github.com/ipfs/go-cid"
+	"golang.org/x/xerrors"
+
+	"github.com/go-pg/pg/v10"
+	"go.opentelemetry.io/otel/api/global"
+	"go.opentelemetry.io/otel/api/trace"
+	"go.opentelemetry.io/otel/label"
+)
+
+func NewProcessingStateChange(ts *types.TipSet) *ProcessingStateChange {
+	return &ProcessingStateChange{
+		TipSet:  ts.Key().String(),
+		Height:  int64(ts.Height()),
+		AddedAt: time.Now(),
+	}
+}
+
+type ProcessingStateChange struct {
+	tableName struct{} `pg:"visor_processing_statechanges"`
+
+	TipSet string `pg:",pk,notnull"`
+
+	Height int64 `pg:",use_zero"`
+
+	// AddedAt is the time the block was discovered and written to the table
+	AddedAt time.Time `pg:",notnull"`
+
+	// ClaimedUntil marks the block as claimed for processing until the set time
+	ClaimedUntil time.Time
+
+	// CompletedAt is the time the block was read from the chain and analysed for actor state changes
+	CompletedAt time.Time
+
+	// ErrorsDetected contains any error encountered when analysing the block for actor state changes
+	ErrorsDetected string
```

Comment:
Can this be made nullable? It makes checking for whether an error has occurred simpler

Reply:
hmm it seems it is but we are setting it to an empty string at some point

Reply:
Done, will write null when marking complete with no error message